### DISCUSSION TITLE
v17.0.0.0beta2

### DISCUSSION THREAD

**Discussion by awawa-dev** [awawa-dev/HyperHDR#115] [General]
Changes from the previous v17 beta:
- Fix: serious bug when the WWW panel after the restart had first instance's configuration
- New feature: auto-resume option for video grabbers when the stream stops for some reason (for example: changing the resolution or refresh rate of the incoming video stream) 
- New feature: add support for ESP32 devices for HyperSPI (awa_spi driver)
- New feature: add list of available COM ports for the adalight driver
- New feature: execute multiple JSON commands via GET request using single link
- Fix: in specific cases some devices could not react to 'no-signal' when it's triggered
- Fix: better fit for mobile browsers the remote and JSON API wizard tab pages
- Upgrade: rpi_ws281x library
- Fix: validate user color selection in the remote and JSON API wizard tab pages
  
-----
  
This is a beta version intended for testers or users who want to try out new functionalities.  
It has just come out of production and despite our best efforts it may still contains bugs.  
More detailed information can be found on the HyperHDR blog: [link](https://hyperhdr.blogspot.com/2021/08/hyperhdr-v17beta-preview.html).
**BEFORE INSTALLING PLEASE BACKUP YOUR HYPERHDR CONFIGURATION FOLDER (JUST IN CASE, YOUR CONFIGURATION SHOULD BE PRESERVED)** 

SD card images will be available for the stable release.  
Please put your feedback in dedicated thread in the discussions [link](https://github.com/awawa-dev/HyperHDR/discussions/102) or report a new issue with full details.  
 
  
Change log:
- upgrade ancient Bootstrap 3 to version 5
- replace and update most of UI components, new layout [picture1](https://i.postimg.cc/9QXhFkjt/white1.jpg), [picture2](https://i.postimg.cc/wB7YDsM1/white2.jpg), [picture3](https://i.postimg.cc/Mps8cY6k/white3.jpg), [picture4](https://i.postimg.cc/SNh4HsTP/black1.jpg), [picture5](https://i.postimg.cc/RVLmmB2Z/black2.jpg), [picture6](https://i.postimg.cc/d0pP6tZ4/black3.jpg)
- software screen grabbers (Windows:DirectX11 / Linux:X11 / macOS:CoreGraphics) [pictureWin](https://i.postimg.cc/SNrB5GzX/screen-dx.jpg)
[pictureLin](https://i.postimg.cc/y88MzSFL/screen-x11.jpg) [pictureMac](https://i.postimg.cc/ZKwKyQhH/screen-macos.png) #46
- automatic signal detection with smart learning capability for USB grabbers (you can have only one algorithm enabled at the time: automatic one or the old, difficult to setup and  unreliable manual one) [picture1](https://i.postimg.cc/zfW0mJPd/automatic1.jpg) [picture2](https://i.postimg.cc/JnLKM9fR/automatic2.jpg) [picture3](https://i.postimg.cc/63fc3MKn/automatic3.jpg) [picture4](https://i.postimg.cc/4y81vnm3/automatic4.jpg) [picture5](https://i.postimg.cc/5y93F89G/automatic5.jpg) [picture6](https://i.postimg.cc/G3zzngq1/automatic6.jpg)
- re-implemented backup import / export function for ALL instances settings in place of the old functionality that clumsily imitated creating of backups, which in many cases resulted in data loss [picture1](https://i.postimg.cc/j2jHjKZY/backup.jpg)
- current video stream information in the 'overview' tab [picture1](https://i.postimg.cc/tCw39K68/overview.jpg)
- support for my new HyperSPI project (https://github.com/awawa-dev/HyperSPI) with awa_spi LED driver [picture1](https://postimg.cc/sMVB09LC)
- new video stream crop method in JSON API [picture1](https://i.postimg.cc/ZqFpF0Qh/crop.jpg) #64 
- JSON API documentation in a form of live playground in 'Advanced' tab [picture1](https://i.postimg.cc/RFp3XtQ2/api.jpg)
- LED grouping aka PC mode aka gradient mode, can help with eye fatigue when used with the monitor, each LED in the group has same average color [picture1](https://i.postimg.cc/pV9Q3VhK/gradient.jpg), you can even improve the effect with the anti-flickering filter with threshold set to 255
- add timeout for the anti-flickering filter [picture1](https://i.postimg.cc/Gt04WCVj/timeout.jpg)
- translation resources are updated
- new panel for easy video resolution & refresh mode selection [picture1](https://i.postimg.cc/d0xk9d9p/button.jpg) [picture2](https://i.postimg.cc/cHtKkH9H/panel.jpg)
- add release for AARCH64 architecture #68 **PERFORMANCE INCREASED UP to 30% COMPARED to Rpi 32bits**: [32bit 1280](https://i.postimg.cc/Kz1c5MC1/rpi3-32bit-1280-720-17ms.png) [32bit 1920](https://i.postimg.cc/sfZDQ5p8/rpi3-32bit-1920-1080-38ms.png) [64bit 1280](https://i.postimg.cc/XqFVR0j3/rpi3-64bit-1280-720-12ms.png) [64bit 1920](https://i.postimg.cc/Px0XVqk2/rpi3-64bit-1920-1080-26ms.png)
- fix for WLED new network protocol #90
- HDR tone mapping is available as a global component [picture1](https://i.postimg.cc/t4g4mJ7j/compHDR1.jpg) [picture2](https://i.postimg.cc/RZbZrmnn/compHDR2.jpg) [picture3](https://i.postimg.cc/6Q25tPdb/compHDR3.jpg)
- removed redundant default JSON config in favour of JSON Schemas defaults
- fix missing Linux taskbar icon
- support for libCEC 6.0.2 to turn on/off video & system grabber [picture1](https://i.postimg.cc/NfM33H8p/control.jpg)
- support for libCEC to turn on/off HDR tone mapping with remote buttons (CEC or libCEC is quite buggy here and dependent on the hardware, with LG C9 must restart TV to make them work) [picture1](https://i.postimg.cc/gkTCm2Nd/buttons.jpg)
- HyperHDR is ready for QT6.2 (tested with the preview version/Vulkan/Windows)
- lower CPU usage when automatic signal detection triggers 'nosignal' ('save resources' for software framerate decimation) [picture1](https://i.postimg.cc/wMPw87HG/save.jpg) #93
- standardize libJPEG-turbo library (where it's necessary)
- fix values premature clipping in the LUT generator & SDR preview rendering fix, access available now from the menu ('Advanced' tab)
- suppress most of external components' warnings while building
- faster image to LED colors transformation
- import 'sparks' and 'system shutdown' effects to the new effect API #75
- better logging with instances' indexes
- Fixed power saving issue in macOS version
- fix memory leaks in SPI drivers

<hr /><em>This discussion was created from the release <a href='https://github.com/awawa-dev/HyperHDR/releases/tag/v17.0.0.0beta2'>v17.0.0.0beta2</a>.</em>

**Comment by sofakng** [comment]
How can I install this on a Raspberry Pi 4?  I don't see any SD card images.

Thanks for any help!

**Reply by awawa-dev** [reply]
> SD card images will be available for the stable release. 😉 

But don't worry: it will be release real soon if nothing bad happens.

**Comment by sofakng** [comment]
Ahhh, OK.  I just got everything setup with Hyperion but was noticing a bit of latency so wanted to try out your fork (HyperHDR) with all of it's improvements.

I'll be patient.  Thanks :)

**Reply by awawa-dev** [reply]
Remember to start from fresh settings, lower delay and refresh rate in the smoothing if necessary (depends on your hardware LED strip driver performance), avoid 1080p for capturing (or use 'quarter frame' mode) and everything should be OK 😄

**Comment by sofakng** [comment]
Thanks - I'll give that a try for now.

If I'm not using HDR, do you suggest I stick with Hyperion.ng or "upgrade"
to HyperHDR after the next release?

**Reply by awawa-dev** [reply]
v17 probably will be available probably in one-two weeks: it brings over 30 new features and bug fixing (including fix for LUT HDR/SDR that missed beta2). So I would wait a bit and start from fresh settings: the configuration file is different now from Hyperion NG one, some settings should be imported but such "upgrade" is not recommended. Safer option is always to have a working backup: write HyperHDR's image on second SD card and leave Hyperion SD card just in case.

**Reply by awawa-dev** [reply]
As you can see HDR is just one of HyperHDR's feature: it began from HDR but then it went much further.
